peps/pep-9999.rst

```diff
@@ -0,0 +1,312 @@
+
+A ``NameError`` would make the mistake abundantly clear. Moreover, text editors
+could highlight this based on static analysis ‒ f(x=) is exactly equivalent to
+writing ``f(x=x)``. If ``x`` does not exist, editors currently have no problem
+highlighting that issue.
+
+Recommendations
+================================================
+As with any other language feature, the programmer should be judicious about
+whether using this feature  improves their code based on the context. We do not
+advocate for enforcing a rule of using this feature in all cases where it may be
+applicable.
+
+Reference Implementation
+================================================
+A proposed implementation for
+`cpython is here <https://github.com/Hels15/cpython/tree/last-build>`_.
+
+References
+================================================
+
+.. [1] Issue 36817: Add = to f-strings for easier debugging. - Python tracker
+   https://bugs.python.org/issue36817
+.. [2] Ruby keyword argument syntactic sugar
+   https://www.ruby-lang.org/en/news/2021/12/25/ruby-3-1-0-released/#:~:text=Other%20Notable%20New%20Features
+.. [3] ReasonML named argument punning
+   https://reasonml.github.io/docs/en/function#:~:text=Named%20argument%20punning
+.. [4] SystemVerilog Implicit Port Connections
+   http://www.sunburst-design.com/papers/CummingsDesignCon2005_SystemVerilog_ImplicitPorts.pdf
+.. [5] OCaml Short notation for variable bindings (let-punning)
+   https://v2.ocaml.org/manual/bindingops.html#ss:letops-punning
+.. [6] JavaScript Object Initializer
+   https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer
+.. [7] Rust Using the Field Init Shorthand
+   https://doc.rust-lang.org/book/ch05-01-defining-structs.html#using-the-field-init-shorthand-when-variables-and-fields-have-the-same-name
+.. [8] Short form for keyword arguments and dicts (2013)
+   https://mail.python.org/archives/list/python-ideas@python.org/thread/SQKZ273MYAY5WNIQRGEDLYTKVORVKNEZ/#LXMU22F63VPCF7CMQ4OQRH2CG6H7WCQ6
+.. [9] Keyword arguments self-assignment (2020)
+   https://mail.python.org/archives/list/python-ideas@python.org/thread/SIMIOC7OW6QKLJOTHJJVNNBDSXDE2SGV/
+.. [10] Shorthand notation of dict literal and function call (2020)
+   https://discuss.python.org/t/shorthand-notation-of-dict-literal-and-function-call/5697/1
+.. [11] Allow identifiers as keyword arguments at function call site (extension
+   of PEP 3102?) (2023)
+   https://discuss.python.org/t/allow-identifiers-as-keyword-arguments-at-function-call-site-extension-of-pep-3102/31677
+.. [12] Shorten Keyword Arguments with Implicit Notation: foo(a=a, b=b) to foo(.a, .b) (2023)
+   https://discuss.python.org/t/shorten-keyword-arguments-with-implicit-notation-foo-a-a-b-b-to-foo-a-b/33080
+.. [13] Syntactic sugar to encourage use of named arguments (2023)
+   https://discuss.python.org/t/syntactic-sugar-to-encourage-use-of-named-arguments/36217
```

Comment:
Where appropriate, it would be better to move these references into inline links.

Two benefits:

* Saves users a click
* They can be a pain to maintain/renumber as the text evolves.